stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt

```diff
@@ -116,12 +118,36 @@ public class ChannelListView @JvmOverloads constructor(
         scrollListener.setPaginationEnabled(enabled)
     }
 
-    public fun reachedEndOfChannels(endReached: Boolean) {
-        requireAdapter().endReached = endReached
+    public fun setChannels(channels: List<ChannelListItem>) {
+        requireAdapter().submitList(channels)
     }
 
-    public fun setChannels(channels: List<Channel>) {
-        requireAdapter().submitList(channels)
+    public fun showLoadingMore(show: Boolean) {
+        requireAdapter().let { adapter ->
+            val currentList = adapter.currentList
+            val mutableCopy = currentList.toMutableList()
+
+            val showLoadingMore = show && !loadingMore
+            val hideLoadingMore = !show && loadingMore
+
+            when {
+                showLoadingMore -> {
+                    loadingMore = true
+                    mutableCopy.add(ChannelListItem.LoadingMoreItem)
+                }
+
+                hideLoadingMore -> {
+                    loadingMore = false
+                    mutableCopy.remove(ChannelListItem.LoadingMoreItem)
```

Comment:
When adding/removing this item, we could use [`filterIsInstance`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-is-instance.html) to make sure that we never end up with multiple loading items in the list.